controllers/auth.js

```diff
@@ -77,12 +169,28 @@ const githubAuthCallback = (req, res, next) => {
       userData = {
         github_id: user.username,
         github_display_name: user.displayName,
+        email: user._json.email,
         github_created_at: Number(new Date(user._json.created_at).getTime()),
         github_user_id: user.id,
         created_at: Date.now(),
         updated_at: null,
       };
 
+      if (!userData.email) {
+        const githubBaseUrl = config.get("githubApi.baseUrl");
+        const res = await fetch(`${githubBaseUrl}/user/emails`, {
+          headers: {
+            Authorization: `token ${accessToken}`,
+          },
+        });
+        const emails = await res.json();
```

Comment:
Will this always get the email, when is it that this will be called?

Reply:
When a user logs in via GitHub OAuth, we may not always get the user's email directly in the initial response, particularly if the user has set their email to private in their GitHub settings. To handle this, I’m fetching the email using the access token via GitHub’s /user/emails endpoint. This ensures we always retrieve the correct email, even if it's not included in the initial OAuth response.

Reply:
So GitHub gives email, by directly calling the above URL?
Can you please add the API reference

Reply:
https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user

Reply:
So, basically, you are saying that during OAuth, if an email is private, it is not shared, and by making an API call, it is shared.
Can you share any reference which confirms this

Also what token will be used to get the emails?

has this been tested?

Reply:
I have tested it and it is working. [Stackover flow issue here explains this](https://stackoverflow.com/questions/35373995/github-user-email-is-null-despite-useremail-scope)